From 10eb1623e167fd147ebd5a78ae5e1628eb3e2190 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:37:21 -0400
Subject: [ethernet] i40e: prep vsi_open logic for non-netdev cases

Message-id: <1409658042-22689-81-git-send-email-sassmann@redhat.com>
Patchwork-id: 90274
O-Subject: [RHEL7.1 PATCH 080/281] i40e: prep vsi_open logic for non-netdev cases
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Shannon Nelson <shannon.nelson@intel.com>

Rearrange the "if netdev" logic slightly to get ready for handling
non-netdev VSIs.

Change-ID: Ia0bfe13d4c994a2351a3c31fe725b75caeb397ee
Signed-off-by: Shannon Nelson <shannon.nelson@intel.com>
Signed-off-by: Catherine Sullivan <catherine.sullivan@intel.com>
Tested-by: Kavindya Deegala <kavindya.s.deegala@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit c22e3c6c791221fc00b56e8a8250fa50f3724d3f)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index bf07d05..2b1a46b 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -4314,24 +4314,27 @@ int i40e_vsi_open(struct i40e_vsi *vsi)
  if (err)
   goto err_setup_rx;
 
- if (!vsi->netdev) {
+ if (vsi->netdev) {
+  snprintf(int_name, sizeof(int_name) - 1, "%s-%s",
+    dev_driver_string(&pf->pdev->dev), vsi->netdev->name);
+  err = i40e_vsi_request_irq(vsi, int_name);
+  if (err)
+   goto err_setup_rx;
+
+  /* Notify the stack of the actual queue counts. */
+  err = netif_set_real_num_tx_queues(vsi->netdev,
+         vsi->num_queue_pairs);
+  if (err)
+   goto err_set_queues;
+
+  err = netif_set_real_num_rx_queues(vsi->netdev,
+         vsi->num_queue_pairs);
+  if (err)
+   goto err_set_queues;
+ } else {
   err = EINVAL;
   goto err_setup_rx;
  }
- snprintf(int_name, sizeof(int_name) - 1, "%s-%s",
-   dev_driver_string(&pf->pdev->dev), vsi->netdev->name);
- err = i40e_vsi_request_irq(vsi, int_name);
- if (err)
-  goto err_setup_rx;
-
- /* Notify the stack of the actual queue counts. */
- err = netif_set_real_num_tx_queues(vsi->netdev, vsi->num_queue_pairs);
- if (err)
-  goto err_set_queues;
-
- err = netif_set_real_num_rx_queues(vsi->netdev, vsi->num_queue_pairs);
- if (err)
-  goto err_set_queues;
 
  err = i40e_up_complete(vsi);
  if (err)
-- 
1.7.1